pkg/provider/resource_beacon_analytics.go

```diff
@@ -0,0 +1,306 @@
+	ProjectId                  string                             `tfsdk:"project_id"`
+	LogsUrl                    *string                            `tfsdk:"logs_url"`
+	BackupRetentionPeriod      types.String                       `tfsdk:"backup_retention_period"`
+	CloudProvider              types.String                       `tfsdk:"cloud_provider"`
+	PgType                     types.String                       `tfsdk:"pg_type"`
+	Password                   types.String                       `tfsdk:"password"`
+	PgVersion                  types.String                       `tfsdk:"pg_version"`
+	PrivateNetworking          types.Bool                         `tfsdk:"private_networking"`
+	AllowedIpRanges            []AllowedIpRangesResourceModel     `tfsdk:"allowed_ip_ranges"`
+	CreatedAt                  types.String                       `tfsdk:"created_at"`
+	MaintenanceWindow          *commonTerraform.MaintenanceWindow `tfsdk:"maintenance_window"`
+	ServiceAccountIds          types.Set                          `tfsdk:"service_account_ids"`
+	PeAllowedPrincipalIds      types.Set                          `tfsdk:"pe_allowed_principal_ids"`
+	PgBouncer                  *PgBouncerModel                    `tfsdk:"pg_bouncer"`
+	Pause                      types.Bool                         `tfsdk:"pause"`
+
+	Timeouts timeouts.Value `tfsdk:"timeouts"`
+}
+
+type beaconAnalyticsResource struct {
+	client *api.BeaconAnalyticsClient
+}
+
+func (bar *beaconAnalyticsResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	bar.client = req.ProviderData.(*api.API).BeaconAnalyticsClient()
+}
+
+func (bar *beaconAnalyticsResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_beacon_analytics"
+}
+
+func (bar *beaconAnalyticsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The beacon analystics cluster resource is used to manage BigAnimal beacon analystics clusters.",
+		// using Blocks for backward compatible
+		Blocks: map[string]schema.Block{
+			"timeouts": timeouts.Block(ctx,
+				timeouts.Opts{Create: true, Delete: true, Update: true},
+			),
+			"allowed_ip_ranges": schema.SetNestedBlock{
+				MarkdownDescription: "Allowed IP ranges.",
+				PlanModifiers:       []planmodifier.Set{plan_modifier.CustomAllowedIps()},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"cidr_block": schema.StringAttribute{
+							MarkdownDescription: "CIDR block.",
+							Required:            true,
+						},
+						"description": schema.StringAttribute{
+							MarkdownDescription: "CIDR block description.",
+							Optional:            true,
+						},
+					},
+				},
+			},
+			"cluster_architecture": schema.SingleNestedBlock{
```

Comment:
currently only "single" architecture type is supported for AC, need to check if we will support other archs in future

Reply:
currently this can support "single" and "ha". If it wants to support some brand new architecture or PGD we could add those fields in and change the currently required fields to optional and do a minor release. So as it stands I think it is backwards compatible

Reply:
I found I have removed this. Single is what we currently have and I think we should only support Single hardcoded. If further architectures are supported we can add this in again and call it a feature enhancement. This will stop users trying different cluster architectures